OpenRA.Mods.Common/Projectiles/Missile.cs

```diff
@@ -145,7 +142,22 @@ public class MissileInfo : IProjectileInfo
 			"not trigger fast enough, causing the missile to fly past the target.")]
 		public readonly WDist CloseEnough = new WDist(298);
 
+		[Desc("Scan radius for victims beyond projectile width. If set to zero (default), it will automatically scale to the largest health shape.",
+			"Custom overrides should not be necessary under normal circumstances.")]
+		public WDist TargetExtraSearchRadius = WDist.Zero;
+
 		public IProjectile Create(ProjectileArgs args) { return new Missile(this, args); }
+
+		public void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			var validActors = rules.Actors.Where(a => a.Value.TraitInfos<HealthInfo>().Any()).ToList();
+
+			// TODO: Make this handle multiple Health traits per actor
+			var largestHealthRadius = validActors.Max(a => a.Value.TraitInfo<HealthInfo>().Shape.OuterRadius);
```

Comment:
Indeed - extract a method and save yourself the effort of changing all of the copy-paste code.